dashboard/app/controllers/learning_goal_evaluations_controller.rb

```diff
@@ -0,0 +1,38 @@
+class LearningGoalEvaluationsController < ApplicationController
+  authorize_resource
+
+  def create
+    learning_goal_evaluation = LearningGoalEvaluation.new(learning_goal_evaluation_params)
+    learning_goal_evaluation.teacher_id = current_user.id
+
+    learning_goal_evaluation.save!
+
+    render json: learning_goal_evaluation
+  end
+
+  def update
+    params.require(:id)
+    learning_goal_evaluation = LearningGoalEvaluation.find_by_id(params[:id])
```

Comment:
`find_by_id` will return `nil` if that record doesn't exist. You're not checking for `nil` on the next line, so you'll hit an unhandled exception there in this case. 

A clean way to handle this would be to change the next line (`if learning_goal_evaluation.teacher_id...`) to `if learning_goal_evaluation&.teacher_id...` using the safe navigation operator, since you're already returning a `:not_found` in the `else` block there.